src/Apps/IdentityVerification/IdentityVerificationApp.tsx

```diff
@@ -1,35 +1,131 @@
 import { Box, Button, Sans, Serif } from "@artsy/palette"
 import { IdentityVerificationApp_me } from "__generated__/IdentityVerificationApp_me.graphql"
+import {
+  IdentityVerificationAppStartFlowMutation,
+  IdentityVerificationAppStartFlowMutationResponse,
+} from "__generated__/IdentityVerificationAppStartFlowMutation.graphql"
 import { AppContainer } from "Apps/Components/AppContainer"
 import { useTracking } from "Artsy"
 import * as Schema from "Artsy/Analytics/Schema"
 import { ErrorPage } from "Components/ErrorPage"
-import React from "react"
+import { ErrorModal } from "Components/Modal/ErrorModal"
+import React, { useState } from "react"
 import { Meta, Title as HeadTitle } from "react-head"
-import { createFragmentContainer } from "react-relay"
-import { graphql } from "relay-runtime"
+import {
+  commitMutation,
+  createFragmentContainer,
+  graphql,
+  RelayProp,
+} from "react-relay"
+import createLogger from "Utils/logger"
+// import { RedirectException } from "found"
+
+const logger = createLogger("IdentityVerificationApp.tsx")
 
 interface Props {
   me: IdentityVerificationApp_me
+  relay: RelayProp
+  // match: any
 }
 
-const IdentityVerificationApp: React.FC<Props> = ({ me }) => {
+const IdentityVerificationApp: React.FC<Props> = ({ me, relay }) => {
+  // if (!me) {
+  //   const loginPath =
+  //     "/log_in?redirect_uri=" + encodeURIComponent(match.location.pathname)
+  //   throw new RedirectException(loginPath)
+  // }
   const { identityVerification } = me
+  const { environment } = relay
+
+  const [requesting, setRequesting] = useState(false)
+  const [showErrorModal, setShowErrorModal] = useState(false)
 
   if (!identityVerification || identityVerification.userID !== me.internalID) {
     return <ErrorPage code={404} />
   }
 
   const { trackEvent } = useTracking()
 
-  const clickContinueToVerification = () => {
+  const trackClickedContinueToVerification = () => {
     trackEvent({
       context_page_owner_id: identityVerification.internalID,
       action_type: Schema.ActionType.ClickedContinueToIdVerification,
       context_page: Schema.PageName.IdentityVerificationPage,
     })
   }
 
+  function startIdentityVerification() {
+    const mutation = new Promise<
+      IdentityVerificationAppStartFlowMutationResponse
+    >((resolve, reject) => {
+      commitMutation<IdentityVerificationAppStartFlowMutation>(environment, {
+        onCompleted: data => resolve(data),
+        onError: error => reject({ error }),
```

Comment:
Major question on approach is here. Do we have any very simple examples for a pattern on doing a mutation and handling errors either from metaphysics (in `onCompleted`) or request failure (`onError`)? In psuedocode, my current version looks like this:

```ts
 const mutation = new Promise<
   IdentityVerificationAppStartFlowMutationResponse
 >((resolve, reject) => {
   commitMutation<IdentityVerificationAppStartFlowMutation>(environment, {
     onCompleted: data => resolve(data),
     onError: error => reject({ error }),
     mutation: graphql`mutation {...`
   })
 })
  
const handleMutationSuccess = (
    result: IdentityVerificationAppStartFlowMutationResponse
  ) => {
    const {
      startIdentityVerification: {
        startIdentityVerificationResponseOrError: {
          identityVerificationFlowUrl,
          mutationError,
        },
      },
    } = result
    if (mutationError) {
      handleMutationError(mutationError)
    } else {
      location.assign(identityVerificationFlowUrl)
    }
  }
  const handleMutationError = (mutationError: {
    /** Message/detail from commitMutation conCompleted handler */
    message?: string
    detail?: string
    /** Error from commitMutation onError handler (also possibly in the mutation result) */
    error?: string
  }) => {
    logger.error("FAIL", mutationError)
}

  mutation.then(handleMutationSuccess).catch(handleMutationError)

  }
```

Reply:
See [here](https://github.com/damassi/reaction/blob/d43843bf504b980fa8d6862cc3520c39b9c6065e/src/Apps/Search/Components/SendFeedback.tsx#L90-L92) for a possible approach to handling errors via `ErrorWithMetadata`.

Reply:
(Also lots of different examples in Reaction when searching for `commitMutation`; maybe one of those results will help.)